db/migrate/20210303161124_add_null_constraint_to_agencies_abbreviation.rb

```diff
@@ -0,0 +1,8 @@
+class AddNullConstraintToAgenciesAbbreviation < ActiveRecord::Migration[6.1]
+  def change
+    # following guidance from strong_migration
+    safety_assured do
+      execute 'ALTER TABLE "agencies" ADD CONSTRAINT "agencies_abbreviation_null" CHECK ("abbreviation" IS NOT NULL) NOT VALID'
+    end
```

Comment:
should we do this as separate `up` / `down` migrations?

Reply:
I copied this straight out of the error message from `strong_migrations` (guidance [here](https://github.com/ankane/strong_migrations#setting-not-null-on-an-existing-column)). Unfortunately it looks like this guidance also applies to Postgres 12+, and as far as I'm aware we're still using 9, maybe we just go simpler by just wrapping the `change_column_null` in `safety_assured` since the table only has ~140 elements?

Reply:
Sure, I'm just saying I think we should aim to have reversible migrations when we can, and a plain `execute` statement isn't reversible the way other AR migrations are

All we'd need for a down would probably be `ALTER TABLE "agencies" DROP CONSTRAINT "agencies_abbreviation_null"`

Reply:
Ah ok I see what you're saying. It looks like the functionality still exists in PG 9.6 so I'll tweak the migrations. Thanks!